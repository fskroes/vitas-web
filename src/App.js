import './App.css';
import React from 'react';
import Upload from './component/upload';
import Header from './component/header';
import CssBaseline from '@material-ui/core/CssBaseline';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';
import Footer from './component/footer';

function App() {
  const [isAPIAlive, setApiAlive] = React.useState(false);

  React.useEffect(() => {
    fetch('http://localhost:9000/testAPI')
      .then(res => res.text())
      .then(() => setApiAlive(true))
      .catch(() => setApiAlive(false))
  }, [isAPIAlive]);

  const useStyles = makeStyles((theme) => ({
    root: {
      flexGrow: 1,
    },
    appBarSpacer: theme.mixins.toolbar,
    content: {
      flexGrow: 1,
      height: '100vh',
      overflow: 'auto',
    },
    container: {
      paddingTop: theme.spacing(4),
      paddingBottom: theme.spacing(4),
    }
  }));

  const classes = useStyles();



  if (!isAPIAlive) return <p>API is alive.</p>
  return(
    <React.Fragment>
      <div className={classes.root}>
        <CssBaseline />
        <Header />
        <main className={classes.content}>
          <div className={classes.appBarSpacer} />
          <Container container maxWidth="auto" className={classes.container}>

            <Upload />

          </Container>
        </main>
      </div>
      <Footer title='Footer' description="Something here to give the footer a purpose!"/>
    </React.Fragment>
  )
}

export default App;