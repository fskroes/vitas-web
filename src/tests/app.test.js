import React from 'react';
import {shallow, mount, render} from 'enzyme';
import App from '../App';

it("renders without crashing", () => {
    shallow(<App />);
});

it("renders Account header", () => {
    const wrapper = shallow(<App />);
    const welcome = <p>API is alive.</p>;
    expect(wrapper.contains(welcome)).toEqual(true);
});

it('should render correctly with no props', () => {
    const component = shallow(<App/>);
    
    expect(component).toMatchSnapshot();
});