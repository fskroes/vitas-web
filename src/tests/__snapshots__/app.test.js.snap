// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<Fragment>
  <div
    className="makeStyles-root-5"
  >
    <WithStyles(CssBaseline) />
    <Header />
    <main
      className="makeStyles-content-7"
    >
      <div
        className="makeStyles-appBarSpacer-6"
      />
      <WithStyles(ForwardRef(Container))
        className="makeStyles-container-8"
        container={true}
        maxWidth="auto"
      >
        <Upload />
      </WithStyles(ForwardRef(Container))>
    </main>
  </div>
  <Footer
    description="Something here to give the footer a purpose!"
    title="Footer"
  />
</Fragment>
`;
